### DISCUSSION TITLE
Are there any instructions on how to install Machinaris directly on Ubuntu?

### DISCUSSION THREAD

**Discussion by zorner** [guydavis/machinaris#806] [General]
Running plotman plot in a docker container on Windows is kind of pointless. Docker runs in WSL version 2. WSL v2 has many enhancements, but I/O is limited. Thus plotting that uses a lot of I/O is severely hindered. On my setup, Madmax via Machinaris in a docker container takes 2.5 hours to do a plot. On the native host OS, Windows 10, the same Madmax setup take 35 minutes per plot.

I have tried to find a way to get out of the docker container and over to Ubuntu using WSL v1 that has better I/O. But there dose not seem to be a way to do it. 

I have also tried downgrading docker to Hyper-v, but Machinaris crashes and burns with I/O errors.

Thus I want to try and run Machinaris directly on the Ubuntu using WSL v1 distro. Are there any instructions on how to install Machinaris directly on Ubuntu?

Alternatively as host OS has access to .machinaris what does Plotman need do the analysts? I have a batch script that updates plotman.log and places the plot logs in the plotman\logs directory.

**Comment by guydavis** [comment]
Yes, agreed that Docker i/o performance on Windows (any container) is [very poor](https://github.com/guydavis/machinaris/wiki/Windows#slower-disk-performance).  As such, plotting within Machinaris is recommended only on non-Windows host OSes such as [Linux](https://github.com/guydavis/machinaris/wiki/Linux).  Some users plot manually on Windows directly, and simply point Machinaris (harvester) at the finished plots.

Running Machinaris outside of Docker is **not currently supported**, however you are welcome to tinker.  All working [contributions](https://github.com/guydavis/machinaris/wiki/Contributing) with validation are welcomed.  Thanks!
